src/runtime/ndarray.cc

```diff
@@ -343,6 +347,18 @@ std::shared_ptr<SharedMemory> NDArray::GetSharedMem() const {
   return this->data_->mem;
 }
 
+bool NDArray::IsPinned() const {
```

Comment:
I'm thinking of removing all the `kDLCPUPinned` device type and using `IsPinned()` instead to avoid many workarounds for the context. I can help you with that by sending a PR to this branch in your repo.

Reply:
Sure, please do.

Reply:
Submitted the PR here: https://github.com/BarclayII/dgl/pull/2